packages/api/tests/services/dbt.test.ts

```diff
@@ -0,0 +1,227 @@
+  name: 'dau',
+  description: 'dau',
+  relationName: '',
+  rawSql: `select dt as dt, count(distinct distinct_id) as dau
+  from
+  	(select distinct_id, dt from {{ ref("event") }}) e
+    join
+	(select uid from {{ ref("user") }}) u
+	on u.uid = e.distinct_id`,
+  compiledSql: `select dt as dt, count(distinct distinct_id) as dau
+  from
+  	(select distinct_id, dt from test.events) e
+    join
+	(select uid from test.user) u
+	on u.uid = e.distinct_id`,
+  type: 'model',
+  materialized: 'table',
+}
+
+test.serial('list current dbt blocks', async (t) => {
+  // create blocks first
+  const entities = _([metadata_source_event, metadata_source_user, metadata_model_dau])
+    .map(generateDbtBlock)
+    .value()
+  await getRepository(BlockEntity).save(entities)
+
+  const loadedBlocks = await dbtService.listCurrentDbtBlocks('test')
+
+  t.is(loadedBlocks.length, 3)
+  await getRepository(BlockEntity).delete(_(entities).map('id').value())
+})
+
+test.serial('push exported metadata', async (t) => {
+  // create blocks first
+  const entities = _([metadata_source_event, metadata_model_dau]).map(generateDbtBlock).value()
+  await getRepository(BlockEntity).save(entities)
+
+  const [dbtSourceId, dbtModelId] = _(entities).map('id').value()
+
+  // use operation to create question, to correctly generate links
+  const [anotherParentQuestionId, anotherParentQuestionOp] = generateQuestionOp(
+    'parent',
+    'select 1',
+  )
+  const [children1Id, children1Op] = generateQuestionOp(
+    'children 1',
+    `select * from {{ ${dbtSourceId} }}`,
+  )
+  const [children2Id, children2Op] = generateQuestionOp(
+    'children 2',
+    `select * from {{ ${dbtModelId} }}`,
+  )
+  const [children3Id, children3Op] = generateQuestionOp(
+    'children 3',
+    `select * from {{ ${children1Id} }} join {{ ${anotherParentQuestionId} }}`,
+  )
+
+  await getConnection().transaction(async (manager) => {
+    const op = new BlockOperation(uuid(), 'test', manager)
+    await set(op, anotherParentQuestionId, anotherParentQuestionOp, [])
+    await set(op, children1Id, children1Op, [])
+    await set(op, children2Id, children2Op, [])
+    await set(op, children3Id, children3Op, [])
+  })
+
+  const exportedMeta = await dbtService.loadAllDbtBlockDescendent('test')
+  console.log('exported meta:', exportedMeta)
+  t.is(exportedMeta.length, 3)
+  t.deepEqual(_(exportedMeta).sortBy('name').value(), [
+    {
+      name: 'children_1',
+      sql: "select * from {{ source('event', 'event') }}",
+    },
+    {
+      name: 'children_2',
+      sql: "select * from {{ ref('dau') }}",
+    },
+    {
+      name: 'children_3',
+      sql: `select * from {{ ref('children_1') }} join tellery_transclusion.${anotherParentQuestionId}`,
+    },
+  ])
+  await getRepository(BlockEntity).delete([
+    dbtSourceId,
+    dbtModelId,
+    anotherParentQuestionId,
+    children1Id,
+    children2Id,
+    children3Id,
+  ])
+})
+
+test.serial('update dbt blocks', async (t) => {
```

Comment:
Some test cases are needed to verify the correctness of the bidirectional link

Reply:
For the case that question blocks link to dbt blocks, the correctness of bidirectional links is assured by the correctness of `listAllDbtBlockDescendent`.
Dbt blocks themselves are always the top-level blocks, that is, the source of bidirectional links (they will not link to any other blocks)